Source/Core/DolphinQt2/Config/Mapping/MappingWindow.cpp

```diff
@@ -289,7 +289,22 @@ void MappingWindow::SetMappingType(MappingWindow::Type type)
     widget = new HotkeyGeneral(this);
     AddWidget(tr("General"), widget);
     AddWidget(tr("TAS Tools"), new HotkeyTAS(this));
-    AddWidget(tr("Debugging"), new HotkeyDebugging(this));
+
+    HotkeyDebugging* debugging_widget = new HotkeyDebugging(this);
+    connect(&Settings::Instance(), &Settings::DebugModeToggled, [=](bool enabled) {
+      if (enabled)
+      {
+        m_tab_widget->insertTab(2, GetWrappedWidget(debugging_widget, this, 150, 150),
```

Comment:
You should store the previous widget instead of creating a new one every time, because removeTab does not delete the object -- meaning you'll leak a debugging widget + the wrapper every time.